### DISCUSSION TITLE
Questions about data persistence

### DISCUSSION THREAD

**Discussion by eSPiYa** [danielgerlag/workflow-core#1290] [Q&A]
I'm trying out this workflow core project for microservices orchestration, and it can handle most of possible implementations and scenarios we can think of.

Regarding data persistence, why is it needed? What I can only think of is a feature of auto-resume in case of crashes or restarts, but I can't find any mentions of this. I was able to find a mention of resume but it seems it will only work on suspended workflows. Are interrupted workflows due to crashes or restarts included on this?

And in case I scaled out the orchestrator and used Azure Service Bus or RabbitMQ for queueing, is data persistence required for this?

**Comment by bainewedlock** [comment]
I think the persistence is useful for many reasons, but mainly to remember your unfinished workflows in case of a service restart.
